packages/vite/src/node/utils.ts

```diff
@@ -124,14 +124,15 @@ export function removeImportQuery(url: string) {
 }
 
 export function injectQuery(url: string, queryToInject: string) {
-  let resolvedUrl = new URL(url, 'relative:///')
+  let resolvedUrl = new URL(url.replace(/%/g, '%25'), 'relative:///')
```

Comment:
We can have a comment here to explain why we need this

Reply:
@GrygrFlzr I think we can also link to this PR in the comment (just "#2614" is enough instead of the url)

Reply:
Added both.